src/Core/Scheduler/Providers/PointCloudProvider.js

```diff
@@ -0,0 +1,274 @@
+                        const myname = childname.substr(root.name.length);
+                        url = `${root.baseurl}/${myname}`;
+                    }
+                    const item = { numPoints: n, childrenBitField: c, children: [], name: childname, baseurl: url, bbox: bounds };
+                    snode.children.push(item);
+                    stack.push(item);
+                }
+            }
+        }
+
+        return root;
+    });
+}
+
+// Parse .bin PotreeConverter format
+function readPositionAndColorPotreeBinFormat(ab, numPoints) {
+    const view = new DataView(ab);
+    let offset = 0;
+
+    numPoints = Math.min(numPoints, Math.floor(ab.byteLength / 16));
+
+    const positions = new Float32Array(3 * numPoints);
+    const colors = new Uint8Array(4 * numPoints);
+
+    for (let i = 0; i < numPoints; i++) {
+        positions[3 * i] = view.getUint32(offset + 0, true);
+        positions[3 * i + 1] = view.getUint32(offset + 4, true);
+        positions[3 * i + 2] = view.getUint32(offset + 8, true);
+
+        colors[4 * i] = view.getUint8(offset + 12);
+        colors[4 * i + 1] = view.getUint8(offset + 13);
+        colors[4 * i + 2] = view.getUint8(offset + 14);
+        colors[4 * i + 3] = 255;
+
+        offset += 16;
+    }
+    return [positions, colors];
+}
+
+// Parse .cin PotreeConverter format (see https://github.com/peppsac/PotreeConverter/tree/custom_bin)
+function readPositionAndColorPotreeCinFormat(ab, numPoints) {
+    const positions = new Float32Array(ab, 6 * 4, 3 * numPoints);
+    const icolors = new Uint8Array(ab, 6 * 4 + 3 * 4 * numPoints, 4 * numPoints);
+
+    return [positions, icolors];
+}
+
+
+let nextuuid = 1;
+function pointsFromArrayBuffer(ab, readPositionAndColor, hasBbox) {
+    // hardcoded position + color
+    const numPoints = (ab.byteLength - (hasBbox ? 6 * 4 : 0)) / (3 * 4 + 4 * 1);
+
+    const [positions, colors] = readPositionAndColor(ab, numPoints);
+
+    let tightbbox;
+    if (hasBbox) {
+        const view = new DataView(ab, 0, 6 * 4);
+        const min = new THREE.Vector3(view.getFloat32(0, true), view.getFloat32(4, true), view.getFloat32(8, true));
+        const max = new THREE.Vector3(view.getFloat32(12, true), view.getFloat32(16, true), view.getFloat32(20, true));
+        tightbbox = new THREE.Box3(min, max);
+    } else {
+        const min = new THREE.Vector3(positions[0], positions[1], positions[2]);
+        const max = min.clone();
+        for (let i = 1; i < numPoints; i++) {
+            const v = { x: positions[3 * i], y: positions[3 * i + 1], z: positions[3 * i + 2] };
+            min.min(v);
+            max.max(v);
+        }
+        tightbbox = new THREE.Box3(min, max);
+    }
+
+    var geometry = new THREE.BufferGeometry();
+    geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 4, true));
+
+    // generate unique id for picking
```

Comment:
For factorization purposes, it may be worth it to use the same method for picking (of multiple geometries packed in the same geometry object) that is used in 3d-tiles: a `batch_id` that is unique to each object in the tile and another id unique to the tile.